src/ORM/DataObject.php

```diff
@@ -2447,6 +2448,27 @@ public function getCMSActions()
         return $actions;
     }
 
+    /**
+     * @return ValidatorList
+     */
+    public function getValidatorList(): ValidatorList
```

Comment:
This needs a better Docblock too. Again, we're introducing an API that will be used by 1000s of different sites and we need to be able to make changes to how the core uses it without breaking those sites. No pressure! ;-)

Some things we need to clarify, some of which might prompt adjustments to our API...

 * Is the validator applied here intended to be only used in the CMS/admin context, or is it safe to use for front-end forms as well?
 * How do we define the difference between those two use-cases?
 * If just for the admin context, should it be called `getCMSValidatorList()`?
 * If it's intended to be used multiple contexts, do we need to assume that the validator will be identical in all contexts? If not, should there be a string $context argument that could be e.g. 'cms', 'website', but app developers could also use and handle their own contexts? How should an unknown context be treated?
 * How should people customise this in subclasses? Overriding the method will mean that people need to re-implement the extension code, or face having extensions applied before their subclass overrides – this mirrors getCMSFields / updateCMSFields, but has been a pain there. Perhaps there needs to be a pair of methods:
  * one is `public final` and is the one you call. If contains the invokeWithExtensions() call.
  * one is `protected` and is the one you override. By default it contains the getCMSValidator polyfill.

Related to this although I think CompositeValidator a great class name, I've got more mixed feelings about calling this accessor `getCompositeValidator()`. What do others think? cc @robbieaverill @unclecheese

Reply:
> Is the validator applied here intended to be only used in the CMS/admin context, or is it safe to use for front-end forms as well?

Are `Validator`s in general intended for both? Because the `CompositeValidator` can be used in the same way as a regular `Validator`. It has the same public API, and returns the same (single) `ValidationResult`.

I can definitely update the docblock to reflect whatever the answer to the above is.

> How should people customise this in subclasses?

It is intended to be used in the same way as `getCMSFields`. I feel like that's a paradigm that we're all familiar with. If we're wanting to change that paradigm, then I'm definitely happy to take that on as part of this change - so long as it's going to be mirrored in the future for other areas (like `getCMSFields`, and `getCMSActions`).

Reply:
Validator is available to all Forms. `getCMSValidator()` was used as the companion of `getCMSFields()` and was intended to be applicable to administrative fields only.

**Potentially we should call this `getCMSCompositeValidator()` if we want to maintain the same scope of applicability?**

Regarding subclassing, retaining the same setup as `getCMSFields()` in spite of the same limitations seems okay.

Reply:
Cool. `getCMSCompositeValidator()` sounds good to me, and matches the current convention.

Thanks :)